src/test/java/seedu/address/logic/commands/EditCommandTest.java

```diff
@@ -35,6 +40,88 @@ public class EditCommandTest {
 
     private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
 
+    @Test
+    public void execute_editModuleOldModuleNotFound_throwsCommandException() {
+        Person originalPerson = TypicalPersons.BENSON;
+        Module oldModule = new Module("CS1111");
+        Module newModule = new Module("CS1231S");
+
+        EditPersonDescriptor descriptor = new EditPersonDescriptorBuilder().build();
+        descriptor.setModuleChanges(oldModule, newModule);
+
+        EditCommand editCommand = new EditCommand(originalPerson.getStudentId(), descriptor);
+
+        String expectedMessage = EditCommand.MESSAGE_MODULE_NOT_FOUND;
+        assertThrows(CommandException.class, () -> editCommand.execute(model), expectedMessage);
+    }
+    @Test
+    public void execute_editModuleWithoutGrade_success() throws Exception {
+        Person originalPerson = TypicalPersons.BENSON;
+        Module oldModule = new Module("GEC1044");
+        Module newModule = new Module("CS1231S");
+
+        Person expectedPerson = new PersonBuilder()
+                .withStudentId("19191919")
+                .withName("Benson Meier")
+                .withAddress("311, Clementi Ave 2, #02-25")
+                .withEmail("johnd@example.com")
+                .withPhone("98765432")
+                .withCourse("Medicine")
+                .withTag("Student")
+                .addUngradedModule("CS1231S")
+                .build();
+
+        EditPersonDescriptor descriptor = new EditPersonDescriptorBuilder().build();
+        descriptor.setModuleChanges(oldModule, newModule);
+
+        EditCommand editCommand = new EditCommand(originalPerson.getStudentId(), descriptor);
+
+        Model expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.setPerson(originalPerson, expectedPerson);
+
+        String expectedMessage = String.format(EditCommand.MESSAGE_EDIT_PERSON_SUCCESS,
+                Messages.format(expectedPerson));
+
+        assertCommandSuccess(editCommand, model, expectedMessage, expectedModel);
+
+        assertTrue(expectedPerson.getModules().stream().anyMatch(m -> m.value.equals("CS1231S")));
+        assertFalse(expectedPerson.getModules().stream().anyMatch(m -> m.value.equals("GEC1044")));
+    }
+
+    @Test
```

Comment:
Good addition of test cases to provide comprehensive testing to the new additions to the EditCommand